src/main/java/org/radarbase/appserver/controller/RadarProjectController.java

```diff
@@ -120,16 +120,36 @@ public ResponseEntity<ProjectDto> addProject(
    *     org.radarbase.appserver.exception.NotFoundException} if project was not found.
    */
   @Authorized(
-      permission = AuthPermissions.CREATE,
-      entity = AuthEntities.MEASUREMENT,
+      permission = AuthPermissions.UPDATE,
+      entity = AuthEntities.SUBJECT,
       permissionOn = PermissionOn.PROJECT)
   @PutMapping(
       value = "/" + PathsUtil.PROJECT_PATH + "/" + PathsUtil.PROJECT_ID_CONSTANT,
       consumes = {MediaType.APPLICATION_JSON_VALUE})
   public ResponseEntity<ProjectDto> updateProject(
-      @Valid @PathParam("projectId") String projectId, @Valid @RequestBody ProjectDto projectDto) {
-    ProjectDto projectDto1 = this.projectService.updateProject(projectDto);
-    return ResponseEntity.ok(projectDto1);
+      @Valid @PathParam("projectId") String projectId, @Valid @RequestBody ProjectDto projectDto,
+      HttpServletRequest request) {
+
+    if (authorization != null) {
+      RadarToken token = (RadarToken) request.getAttribute(AuthAspect.TOKEN_KEY);
+      if (authorization.hasPermission(
+              token,
+              AuthPermissions.UPDATE,
+              AuthEntities.SUBJECT,
+              PermissionOn.PROJECT,
+              projectDto.getProjectId(),
+              null,
+              null)) {
+        ProjectDto projectDto1 = this.projectService.updateProject(projectDto);
+        return ResponseEntity.ok(projectDto1);
```

Comment:
you don't need this when using `PermissionOn.PROJECT` and the projectId is available in path params. This is only needed when projectId is not part of path param.